src/components/Game/Game.js

```diff
@@ -19,17 +20,40 @@ class GameView extends Component {
         letters: [],
         lettersInSlots: [],
         isSubmitVisible: false,
+        loading: true,
+        isResponseOK: false,
     };
 
     componentDidMount() {
-        const { dispatch } = this.props;
-        dispatch(initWords());
+        const { allWords, dispatch } = this.props;
+        if (allWords.length) {
+            dispatch(initWords(allWords));
+        } else {
+            dispatch(getAllWords());
```

Comment:
it seems this action is dispatched in `App`
`App` component is a root component, its `componentDidMount` is always run

Reply:
the above should answer your comment `I still had to add making a request in Game`
Game is a child of App

Reply:
did you try to remove `dispatch(getAllWords())` from Game and do what I described: 

> in cases a user goes directly to /game page or refreshes it

?

when you do it the App is rerendered. so the sequence is: `App` renders -> `Game` mounts -> `componentDidMount` in `App` is called. so `Game` mounts before data is fetched (and `initWords` in `Game` is synchronous).

Reply:
I just did as instructed! what's the conclusion?

In general my expectation (something I consider to be a good practice) is that a certain action/operation should be done/initialized/dispatched in one place only if it's possible. This is what I tried to express. Here in our app I have noticed `getAllWords` being dispatched in two places so I got suspicious. I don't see a reason for that.

Maybe I don't understand something! Then please try to explain and correct my thinking :)
Otherwise please consider my comment or perhaps ask questions…